### PR TITLE
Add mit and grid planck layouts

### PR DESCRIPTION
**Pull request by renzmann**
Creating this PR for some early feedback. I use two variants of the planck keyboard, one with the "MIT" space bar and one that's a full 4x12 grid layout that I update pretty frequently, so `keymapviz` is a really helpful tool.

Still in progress is adding some test cases under the "test" directory. The default layout is a bit weird, since [they use a grid keymap.c file](https://github.com/qmk/qmk_firmware/blob/485cf920b4f2d7bf5702573d008c5e435ee570a2/keyboards/planck/keymaps/default/keymap.c#L45-L61), but this still corresponds to the MIT layout in their artwork.

### CODE REVIEW

**Comment by renzmann** [comment]
I was able to find the [rootiest](https://github.com/qmk/qmk_firmware/tree/485cf920b4f2d7bf5702573d008c5e435ee570a2/keyboards/planck/keymaps/rootiest) layout which uses the `LAYOUT_planck_mit` keyword for testing the mit layout, and just accepting that the default would have a grid appearance.

File: keymapviz/keyboards/planck/__init__.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+
+keymap_keyword = '(?:LAYOUT_planck_grid|LAYOUT_planck_mit|LAYOUT_ortho_4x12|LAYOUT_ortho_1x2u)'
```
Comment (yskoht) [inline keymapviz/keyboards/planck/__init__.py:3]:
```suggestion
keymap_keyword = '(?:LAYOUT_planck_grid|LAYOUT_planck_mit|LAYOUT_ortho_4x12|LAYOUT_planck_1x2uC)'
```

I couldn't find `LAYOUT_ortho_1x2u` , so I thought `LAYOUT_planck_1x2uC` would be correct.

File: keymapviz/keyboards/planck/__init__.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+layout_editor_json = {
```
Comment (yskoht) [inline keymapviz/keyboards/planck/__init__.py:4]: `default` key is required.
I think it should be the same as `grid`.

```
'default': 'keyboards/planck/layout_editor/grid.json',
```

File: keymapviz/keyboards/planck/__init__.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    'grid': 'keyboards/planck/layout_editor/grid.json',
+    'mit': 'keyboards/planck/layout_editor/mit.json'
+}
+
+ascii_art = {
```
Comment (yskoht) [inline keymapviz/keyboards/planck/__init__.py:9]: This also requires a default key.

File: keymapviz/keyboards/planck/layout_editor/grid.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
+[
```
Comment (yskoht) [inline keymapviz/keyboards/planck/layout_editor/grid.json:1]: The files layout_editor/*.json also needs to have the legend replaced with `{}`. (as well as `ascii_art`).

It looks like the following.

https://github.com/yskoht/keymapviz/blob/develop/keymapviz/keyboards/lets_split/layout_editor/default.json

File: keymapviz/keyboards/planck/layout_editor/mit.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
+[
```
Comment (yskoht) [inline keymapviz/keyboards/planck/layout_editor/mit.json:1]: This file also needs to have the legend replaced with `{}`.

File: test/test.py [hunk 6]
```diff
@@ -226,6 +226,34 @@ def exe(cmd):
 exe('keymapviz -k lets_split ./input/custom_legends_lets_split_keymap.c -c ./input/custom_legends_config.properties -o custom_legends_lets_split.c')
 exe('diff -u expect/custom_legends_lets_split.c custom_legends_lets_split.c')
 
+# planck (grid)
```
Comment (yskoht) [inline test/test.py:229]: ❤️ 

A big thank you for adding test!

**Review by yskoht — COMMENTED** [review]
@renzmann 

I'm sorry for the late reply and thank you for your PR !

I made a few comments.
If it's too hard for you to fix, let me know so I can fix it :)

**Comment by renzmann** [comment]
Thank you for the helpful feedback @yskoht! I'm going to convert this to a draft until I get all the suggested changes in, and I'll mark it as ready for review once I get it buttoned up.

**Review by yskoht — APPROVED** [review]
@renzmann 

Thank you! LGTM:+1:

**Pull request merged**